CMakeLists.txt

```
@@ -1603,6 +1604,92 @@ status("")
+list(GET SYS_INFO 3 OS_NAME)
+list(GET SYS_INFO 4 OS_RELEASE)
+list(GET SYS_INFO 5 OS_VERSION)
+list(GET SYS_INFO 6 OS_PLATFORM)
+list(GET SYS_INFO 7 PROCESSOR_NAME)
+list(GET SYS_INFO 8 IS_64BIT)
+list(GET SYS_INFO 9 HAS_FPU)
+
+set(HAVE_SSE2_FLAG    FALSE)
+set(HAVE_SSE3_FLAG    FALSE)
+set(HAVE_SSSE3_FLAG   FALSE)
+set(HAVE_SSE4_1_FLAG  FALSE)
+set(HAVE_SSE4_2_FLAG  FALSE)
+set(HAVE_AVX_FLAG     FALSE)
+set(HAVE_AVX2_FLAG    FALSE)
+set(HAVE_NEON_FLAG    FALSE)
+
+# additional CPU info
+if(X86 OR X86_64 OR WINRT)
+  if(MSVC)
+    if(NOT MSVC64)
+      vp_check_compiler_flag(CXX "/arch:SSE2"   HAVE_SSE2_FLAG    "${PROJECT_SOURCE_DIR}/cmake/checks/cpu_sse2.cpp")
+    endif()
+
+    vp_check_compiler_flag(CXX "/arch:AVX"    HAVE_AVX_FLAG     "${PROJECT_SOURCE_DIR}/cmake/checks/cpu_avx.cpp")
+    vp_check_compiler_flag(CXX "/arch:AVX2"   HAVE_AVX2_FLAG    "${PROJECT_SOURCE_DIR}/cmake/checks/cpu_avx2.cpp")
+    if(${HAVE_AVX_FLAG})
+      set(HAVE_SSE2_FLAG    TRUE)
+      set(HAVE_SSE3_FLAG    TRUE)
+      set(HAVE_SSSE3_FLAG   TRUE)
+      set(HAVE_SSE4_1_FLAG  TRUE)
+      set(HAVE_SSE4_2_FLAG  TRUE)
+    endif()
+  else()
+    vp_check_compiler_flag(CXX "-msse2"   HAVE_SSE2_FLAG    "${PROJECT_SOURCE_DIR}/cmake/checks/cpu_sse2.cpp")
+    vp_check_compiler_flag(CXX "-msse3"   HAVE_SSE3_FLAG    "${PROJECT_SOURCE_DIR}/cmake/checks/cpu_sse3.cpp")
+    vp_check_compiler_flag(CXX "-mssse3"  HAVE_SSSE3_FLAG   "${PROJECT_SOURCE_DIR}/cmake/checks/cpu_ssse3.cpp")
+    vp_check_compiler_flag(CXX "-msse4.1" HAVE_SSE4_1_FLAG  "${PROJECT_SOURCE_DIR}/cmake/checks/cpu_sse41.cpp")
+    vp_check_compiler_flag(CXX "-msse4.2" HAVE_SSE4_2_FLAG  "${PROJECT_SOURCE_DIR}/cmake/checks/cpu_sse42.cpp")
+    vp_check_compiler_flag(CXX "-mavx"    HAVE_AVX_FLAG     "${PROJECT_SOURCE_DIR}/cmake/checks/cpu_avx.cpp")
+    vp_check_compiler_flag(CXX "-mavx2"   HAVE_AVX2_FLAG    "${PROJECT_SOURCE_DIR}/cmake/checks/cpu_avx2.cpp")
+  endif()
+elseif(ARM OR AARCH64)
+  set(HAVE_NEON_FLAG    TRUE)
+endif()
+
+status("")
+status("  System information:")
+status("    Number of CPU logical cores:"                  "${NUMBER_OF_LOGICAL_CORES}")
+status("    Number of CPU physical cores:"                 "${NUMBER_OF_PHYSICAL_CORES}")
+status("    Total physical memory (in MiB):"               "${TOTAL_PHYSICAL_MEMORY}")
+status("    OS name:"                                      "${OS_NAME}")
+status("    OS release:"                                   "${OS_RELEASE}")
+status("    OS version:"                                   "${OS_VERSION}")
+status("    OS platform:"                                  "${OS_PLATFORM}")
+status("    CPU name:"                                     "${PROCESSOR_NAME}")
+status("    Is the CPU 64-bit?"                            ${IS_64BIT} THEN "yes" ELSE "no")
+status("    Does the CPU have FPU?"                        ${HAS_FPU} THEN "yes" ELSE "no")
+status("    SSE2?"                                         ${HAVE_SSE2_FLAG} THEN "yes" ELSE "no")
```

Maybe it should rather be?

```suggestion
status("  Compiler flags support:")
status("    SSE2 flag?"                                         ${HAVE_SSE2_FLAG} THEN "yes" ELSE "no")
status("    SSE3 flag?"                                         ${HAVE_SSE3_FLAG} THEN "yes" ELSE "no")
status("    SSSE3 flag?"                                        ${HAVE_SSSE3_FLAG} THEN "yes" ELSE "no")
status("    SSE4.1 flag?"                                       ${HAVE_SSE4_1_FLAG} THEN "yes" ELSE "no")
status("    SSE4.2 flag?"                                       ${HAVE_SSE4_2_FLAG} THEN "yes" ELSE "no")
status("    AVX flag?"                                          ${HAVE_AVX_FLAG} THEN "yes" ELSE "no")
status("    AVX2 flag?"                                         ${HAVE_AVX2_FLAG} THEN "yes" ELSE "no")
status("    NEON flag?"                                         ${HAVE_NEON_FLAG} THEN "yes" ELSE "no")
```

If `vp_check_compiler_flag` only check for correct compilation result and does not execute the code, it is indeed `Compiler flags support`.